app/src/main/java/cm/aptoide/pt/ads/IronSourceAdRepository.java

```diff
@@ -0,0 +1,87 @@
+package cm.aptoide.pt.ads;
+
+import android.app.Activity;
+import android.util.Log;
+import cm.aptoide.pt.BuildConfig;
+import com.ironsource.mediationsdk.IronSource;
+import com.ironsource.mediationsdk.logger.IronSourceError;
+import com.ironsource.mediationsdk.sdk.InterstitialListener;
+import rx.subjects.PublishSubject;
+
+public class IronSourceAdRepository {
+  private static final String TAG = "IronSourceAdRepository";
+  private final Activity activity;
+  private final PublishSubject<Void> impressionSubject;
+  private final PublishSubject<Void> clickSubject;
+
+  private boolean showInterstitial;
+
+  public IronSourceAdRepository(Activity activity) {
+    this.activity = activity;
+    this.clickSubject = PublishSubject.create();
+    this.impressionSubject = PublishSubject.create();
+    this.showInterstitial = false;
+  }
+
+  public void initialize() {
+    IronSource.init(activity, BuildConfig.IRONSOURCE_APPLICATION_ID);
+    IronSource.setInterstitialListener(new InterstitialListener() {
+      @Override public void onInterstitialAdReady() {
+        if (showInterstitial) {
+          showInterstitialAd();
+          showInterstitial = false;
+        }
+      }
+
+      @Override public void onInterstitialAdLoadFailed(IronSourceError ironSourceError) {
+        Log.i(TAG, "Interstitial Ad failed to load. Reason: " + ironSourceError.getErrorMessage());
```

Comment:
Should not be using Log but Logger.getInstance() since it has logic to disable in release and to be controlled by the Toolbox app.